File: Ejercicio_03/Domain/Interfaces/Departamento/IDepartamentoUseCases.cs
using Domain.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Domain.Interfaces
{
    public interface IDepartamentoUseCases
    {
        List<Departamento> getDepartamentos();
        Departamento getDepartamento(int id);
        int deleteDepartamento(int id);
        int updateDepartamento(int id, Departamento departamento);
        int addDepartamento(Departamento departamento);
    }
}

File: Angular/CRUD/src/domain/interfaces/IDepartamentoUseCases.ts
import { Departamento } from "../entities/Departamento"

export interface IDepartamentoUseCases {
    getDepartamentos(): Array<Departamento>
	getDepartamento(id: number): Departamento
	updateDepartamento(id: number, departamento: Departamento): number
	addDepartamento(departamento: Departamento): number
	deleteDepartamento(id: number): number
}
